examples/dial/example_dial.cpp

```diff
@@ -0,0 +1,63 @@
+#include <grpcpp/client_context.h>
+#include <grpcpp/grpcpp.h>
+#include <robot/v1/robot.grpc.pb.h>
+#include <robot/v1/robot.pb.h>
+#include <unistd.h>
+
+#include <boost/optional.hpp>
+#include <cstddef>
+#include <iostream>
+#include <module/service.hpp>
+#include <ostream>
+#include <robot/client.hpp>
+#include <robot/service.hpp>
+#include <rpc/dial.hpp>
+#include <string>
+#include <vector>
+
+using viam::robot::v1::Status;
+using Viam::SDK::Credentials;
+using Viam::SDK::Options;
+
+extern "C" void* init_rust_runtime();
+extern "C" int free_rust_runtime(void* ptr);
+extern "C" void free_string(char* s);
+extern "C" char* dial(const char* uri, const char* payload, bool allow_insecure, void* ptr);
+
+int main() {
+    const char* uri = "<your robot URI here>";
+    Viam::SDK::DialOptions dial_options = Viam::SDK::DialOptions();
+    std::string payload = "<your payload here>";
+    Credentials credentials(payload);
+    dial_options.credentials = credentials;
+    boost::optional<Viam::SDK::DialOptions> opts(dial_options);
+    std::string address(uri);
+    Options options = Options(1, opts);
```

Comment:
Lots to say about this block. C++ doesn't require you to make so many named objects to achieve what you are actually doing here. I expect the following will work:

```
    std::string address = "<your robot URI here>";
    std::string payload = "<your payload here>";

    DialOptions dial_options;
    dial_options.credentials.emplace(std::move(payload));

    auto robot =
        RobotClient::at_address(
            std::move(address),
            {
                1,
                std::move(dial_options)
            }
        );
```